### PR TITLE
Add preferences

### PR DESCRIPTION
**Pull request by filol**
- Add CheckBox
- Add Switch
- Update gradle
- Change deprecated function inside build.gradle
- Update README
- Update DemoApp for testing preferences items
- Add french translation

#77

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -25,18 +25,18 @@ android {
 }
 
 dependencies {
-    compile fileTree(dir: 'libs', include: ['*.jar'])
-    androidTestCompile('com.android.support.test.espresso:espresso-core:2.2.2', {
+    implementation fileTree(dir: 'libs', include: ['*.jar'])
```
Comment (daniel-stoneuk) [inline app/build.gradle]: Thanks for this!

File: README.md [hunk 2]
```diff
@@ -162,10 +162,12 @@ MaterialAboutCard card = new MaterialAboutCard.Builder()
 
 ### Add Items to a card:
 
-There are currently two types of items you can add to a card - [`MaterialAboutTitleItem`][9] and [`MaterialAboutActionItem`][10]. Other types of items are planned, for example "person" items which feature buttons to showcase a single person. Feel free to submit a PR or Issue for more item ideas.
+There are currently 4 types of items you can add to a card - [`MaterialPreferenceTitleItem`][9], [`MaterialPreferenceActionItem`][10], [`MaterialPreferenceCheckboxItem`][12] and [`MaterialPreferenceSwitchItem`][13]. Other types of items are planned, for example "person" items which feature buttons to showcase a single person. Feel free to submit a PR or Issue for more item ideas.
```
[the remark this answers is not in the document]
Comment (filol) [reply README.md]: Oups

File: library/src/main/java/com/danielstone/materialaboutlibrary/items/MaterialAboutOnCheckedChangedListener.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package com.danielstone.materialaboutlibrary.items;
+
+import android.widget.CompoundButton;
+
+/**
+ * Created by François Dexemple on 04/05/2018
+ */
+public interface MaterialAboutOnCheckedChangedListener {
```
[the remark this answers is not in the document]
Comment (filol) [reply library/src/main/java/com/danielstone/materialaboutlibrary/items/MaterialAboutOnCheckedChangedListener.java]: Yes I will do it but it's not really an action (like click), it's a listener

File: library/src/main/java/com/danielstone/materialaboutlibrary/util/DefaultViewTypeManager.java [hunk 4]
```diff
@@ -1,28 +1,39 @@
 package com.danielstone.materialaboutlibrary.util;
 
 import android.content.Context;
+import android.icu.util.ValueIterator;
```
[the remark this answers is not in the document]
Comment (filol) [reply library/src/main/java/com/danielstone/materialaboutlibrary/util/DefaultViewTypeManager.java]: Nope, I delete it

**Comment by filol** [comment]
I made the corrections. The only details I did not manage to do with Linear layout is to center the switch and the checkbox. You'll see when you put a checkbox under a switch that they are not centered

**Comment by daniel-stoneuk** [comment]
Thanks! I'll check it out and merge onto master branch later

**Comment by filol** [comment]
I forget some error in the README. Check my last commit.

**Pull request merged**